.github/workflows/_go-tests.yml

```diff
@@ -50,39 +50,49 @@ jobs:
 
       # --------------------- DEFAULTS MODE ---------------------
 
-      - name: run tests without race detection and hash state scheme
-        if: matrix.test-mode == 'defaults'
-        id: run-tests-defaults
+      - name: run tests without race detection and hash state scheme (A-batch)
+        if: matrix.test-mode == 'defaults-A'
+        id: run-tests-defaults-a
         continue-on-error: true
         run: >-
           ${{ github.workspace }}/.github/workflows/gotestsum.sh
-          --tags cionly --timeout 60m --test_state_scheme hash --junitfile test-results/junit.xml
+          --tags cionly --timeout 60m --test_state_scheme hash
+          --junitfile test-results/junit-a.xml --run '^Test[A-L]'
```

Comment:
Have you done any analysis to know the actual distribution of tests using these prefixes?
Also, are you sure we don't have any tests which would fail to match both `defaults-A` and `defaults-B` patterns?

I'm just a little concerned that this might be a "fragile" way of splitting them up?

Finally, why are we "only" splitting the `defaults`? Why not also the pathdb? I feel like it runs a long time in our normal CI runs as well.